tests/test_main.py

```diff
@@ -103,3 +104,17 @@ def mocked_main(*args):
 
     with pytest.raises(FakeFailureError):
         xonsh.main.main()
+
+
+def test_xonsh_failback_script_from_file(shell, monkeypatch):
```

Comment:
this is probably due to monkeypatching stderr, `-s` messes with stdout/stderr, can't recall in what way.